processor/elasticinframetricsprocessor/processor.go

```diff
@@ -60,13 +62,49 @@ func (p *ElasticinframetricsProcessor) processMetrics(_ context.Context, md pmet
 	for i := 0; i < md.ResourceMetrics().Len(); i++ {
 		resourceMetric := md.ResourceMetrics().At(i)
 		rm := resourceMetric.Resource()
-
 		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
 			scopeMetric := resourceMetric.ScopeMetrics().At(j)
 			for _, r := range p.remappers {
 				r.Remap(scopeMetric, scopeMetric.Metrics(), rm)
 			}
 		}
 	}
+	// override=True will keep only the metrics that have been remapped based on the presense of OTelRemappedLabel label.
+	// See  https://github.com/elastic/opentelemetry-lib/blob/6d89cbad4221429570107eb4a4968cf8a2ff919f/remappers/common/const.go#L31
+	if p.cfg.Override {
+		newmetic := pmetric.NewMetrics()
+		rmnew := newmetic.ResourceMetrics().AppendEmpty()
+		rmscope := rmnew.ScopeMetrics().AppendEmpty()
+		for i := 0; i < md.ResourceMetrics().Len(); i++ {
+			resourceMetric := md.ResourceMetrics().At(i)
+
+			for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
+				scopeMetric := resourceMetric.ScopeMetrics().At(j)
+				for l := 0; l < scopeMetric.Metrics().Len(); l++ {
+					metric := scopeMetric.Metrics().At(l)
+					if metric.Type().String() == "Gauge" {
+						for m := 0; m < metric.Gauge().DataPoints().Len(); m++ {
+							if oTelRemappedLabel, ok := metric.Gauge().DataPoints().At(m).Attributes().Get(OTelRemappedLabel); ok {
+								if oTelRemappedLabel.Bool() {
+									metric.CopyTo(rmscope.Metrics().AppendEmpty())
```

Comment:
@gizas : BAsically, we go through the whole remapped logic and add the translated and original metrics to a scope metric.
After that you are checking for otel_remapped and creating a new resource/scope metric having only translated metric. 
Can't we have this check in the remap function and in case override is true add only the translated metric to the scopemetric?

Currently we create 2 resourcemetrics/scopemetrics.

Reply:
> Can't we have this check in the remap function and in case override is true add only the translated metric to the scopemetric?

I was trying to do this inside the remap initially. But then the drop has to be implemented in the infraprocessor side. And I believe now is clearer. 

> Can't we have this check in the remap function and in case override is true add only the translated metric to the scopemetric?

Yes this is what I did. I think it works now. I tested the hostemetrics as well and dont see any problems. Just wanted youe eyes on this to see if I miss something. Just another pair of eyes for testting